shared/js/background/classes/sw-tab.es6.js

```diff
@@ -0,0 +1,62 @@
+const Tab = require('./tab.es6')
+
+/**
+ * A stub tab implementation for service workers.
+ *
+ * For most actions, this just mimics a tab with the service worker URL as the tab URL. However, for cases
+ * such as tracker counts, it distributes stats to all known tabs with the same origin.
+ */
+class ServiceWorkerTab extends Tab {
+    /**
+     * @param {string} swUrl
+     * @param {Record<number, Tab>} tabContainer
+     */
+    constructor (swUrl, tabContainer) {
+        super({
+            tabId: -1,
+            url: swUrl,
+            status: null
+        })
+        this.origin = new URL(swUrl).origin
+        this.tabContainer = tabContainer
+    }
+
+    /**
+     * Find the list of tabs which share the same origin as this service worker.
```

Comment:
Shouldn't we be checking for same entity, rather than same origin? I'm wondering if we could compare `tab.site.parentEntity` instead?

Reply:
No - we only want to distribute stats to Tabs which _could have_ initiated this request. Service workers must share the [same origin](https://github.com/w3c/ServiceWorker/blob/main/explainer.md#getting-started) ("The service worker script must be on the same origin as the page") as the sites they operate for, so the origin check should be sufficient.

Reply:
Got it, that makes sense.